### PR TITLE
Throw errors when dots are in fields in targets

### PR DESCRIPTION
**Pull request by Ekrekr**
Fixes https://github.com/dataform-co/dataform/issues/1362#issuecomment-2022561105, which contains the explanation for this solution

Also update filename in utils to basename as it is more correct.

### CODE REVIEW

File: core/actions/assertion.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ export class Assertion extends ActionBuilder<dataform.Assertion> {
     }
 
     if (!config.name) {
-      config.name = Path.fileName(config.filename);
+      config.name = Path.basename(config.filename);
```
Comment (BenBirt) [inline core/actions/assertion.ts:92]: i still think this capitalized import name is super weird (`Path` as opposed to `path` which would be more standard)

File: core/session.ts [hunk 2]
```diff
@@ -392,6 +392,16 @@ export class Session {
       )
     );
 
+    this.checkNoDotsInTargetFields(
```
Comment (BenBirt) [inline core/session.ts:395]: why do we need to do this at the end of compilation? IMO it should be happening well before then - at action declaration time.

Comment (Ekrekr) [reply core/session.ts:395]: Where are you suggesting to put it?

Would it be both:
* Action functions in the session https://github.com/dataform-co/dataform/blob/b5368e04a047c6f2c958b6dd14d9bd5f872e4e0f/core/session.ts#L236C11-L236C21
* Action constructors https://github.com/dataform-co/dataform/blob/b5368e04a047c6f2c958b6dd14d9bd5f872e4e0f/core/actions/table.ts#L291

File: core/actions/assertion.ts (hunk 1 above)
Comment (Ekrekr) [reply core/actions/assertion.ts:92]: It should probably be called `pathUtils` instead. It's capitalized for importing our path.ts file, but lowercase for our node path imports https://github.com/dataform-co/dataform/blob/b5368e04a047c6f2c958b6dd14d9bd5f872e4e0f/core/main_test.ts#L5.

Will update in a subsequent CL.

Comment (BenBirt) [reply core/actions/assertion.ts:92]: IMO it's fine as `path` everywhere, but up to you

File: core/session.ts (hunk 2 above)
Comment (BenBirt) [reply core/session.ts:395]: Everything passes through the action constructors, right? So we should only do it there.

**Review by BenBirt — APPROVED** [review]

**Pull request merged**